### PR TITLE
Implement base filters in search

### PR DESCRIPTION
**Pull request by rafaelkyrdan**
- filter by `minPrice` and `maxPrice`
- Search by location, if there are no coordinates in parameters.

### CODE REVIEW

File: src/app/endpoints/search/search.model.ts [hunk 1]
```diff
@@ -1,14 +1,20 @@
...
-  lat: Joi.string().required(),
-  lng: Joi.string().required(),
+  lat: Joi.string().optional(),
+  lng: Joi.string().optional(),
   from: Joi.date().required(),
-  to: Joi.date().required()  
+  to: Joi.date().required(),
+  minPrice: Joi.number().optional(),
+  maxPrice: Joi.number().optional(),
+  location: Joi.string().optional()
```
Comment (sefabaser) [inline src/app/endpoints/search/search.model.ts]: In joi there is an option to say, its either lat, lng will exist or location will exist, and never together.
https://github.com/hapijs/joi/issues/1234

File: src/app/modules/search/search.ts [hunk 2]
```diff
@@ -3,7 +3,10 @@ import "reflect-metadata";
 import { Supplier, Search } from "../common/interfaces";
 import { SearchItem, Model, Station } from "./search.model";
 import TYPES from "../../config/types"
+import { setFlagsFromString } from "v8";
 const _ = require('lodash');
+const axios = require('axios');
+require('dotenv').config()
```
Comment (sefabaser) [inline src/app/modules/search/search.ts]: what do we use 'dotenv' for?

File: src/app/modules/search/search.ts [hunk 3]
```diff
@@ -59,6 +80,35 @@ class SearchService implements Search {
...
+        }
+      } else {
+        throw new Error("Cannot get the coords for: " + location)
+      }  
+    });
+  }
+
+  private filteryByPrice(searchItems: SearchItem[], minPrice: number, maxPrice: number): SearchItem[] {
+    let filtered = searchItems.filter(item => {
+      return item.price > minPrice && item.price < maxPrice
```
Comment (sefabaser) [inline src/app/modules/search/search.ts]: Can we put equal also?

File: src/app/modules/search/search.ts [hunk 4]
```diff
@@ -36,9 +53,13 @@ class SearchService implements Search {
         return Promise.resolve([])
       }
     })).then(results => {
-      return _.flatten(results)
+      let flattened = _.flatten(results)
+      if (minPrice && maxPrice) {
```
Comment (sefabaser) [inline src/app/modules/search/search.ts]: its actually either or. minPrice might exist but not with maxPrice and vice versa.

File: src/app/modules/search/search.ts [hunk 5]
```diff
@@ -22,8 +25,22 @@ class SearchService implements Search {
...
+      
+    if (!lat && !lng && !location) {
+      throw new Error("Either coords or location should be defined")
+    }
+
+    if (!lat && !lng && location) {
+      let coords = await this.findCoords(location)
+      lat = coords[0]
+      lng = coords[1]
+    }
```
Comment (sefabaser) [inline src/app/modules/search/search.ts]: After placing those validation to Joi there will be no need to check those in here

File: src/app/modules/search/search.ts (hunk 2 above)
Comment (rafaelkyrdan) [reply src/app/modules/search/search.ts]: it is used to get env variable 
`process.env.ACCESS_TOKEN` which is stored in `.env` file

File: src/app/modules/search/search.ts (hunk 4 above)
Comment (rafaelkyrdan) [reply src/app/modules/search/search.ts]: done

**Pull request merged**